packer/build.go

```diff
@@ -79,21 +79,21 @@ type Build interface {
 	SetOnError(string)
 }
 
-// A build struct represents a single build job, the result of which should
+// A CoreBuild struct represents a single build job, the result of which should
 // be a single machine image artifact. This artifact may be comprised of
-// multiple files, of course, but it should be for only a single provider
-// (such as VirtualBox, EC2, etc.).
-type coreBuild struct {
-	name               string
-	builder            Builder
-	builderConfig      interface{}
-	builderType        string
-	hooks              map[string][]Hook
-	postProcessors     [][]coreBuildPostProcessor
-	provisioners       []coreBuildProvisioner
-	cleanupProvisioner coreBuildProvisioner
-	templatePath       string
-	variables          map[string]string
+// multiple files, of course, but it should be for only a single provider (such
+// as VirtualBox, EC2, etc.).
+type CoreBuild struct {
+	Type               string
+	Builder            Builder
+	BuilderConfig      interface{}
+	BuilderType        string
+	Hooks              map[string][]Hook
+	Provisioners       []CoreBuildProvisioner
+	PostProcessors     [][]CoreBuildPostProcessor
+	CleanupProvisioner CoreBuildProvisioner
+	TemplatePath       string
+	Variables          map[string]string
```

Comment:
Here I just made those public in order to configure a build from the hcl2template pkg.